### PR TITLE
AF-529: Fixed NPE when users management is not enabled.

### PR DESCRIPTION
**Pull request by romartin**
Hey @ederign @dgutierr @porcelli 

Do you want to review this? It was just an NPE, but did a bit more refactoring to improve logic and remove a bit of code duplication. 

This is [the JIRA ticket](https://issues.jboss.org/browse/AF-529) for it.

Thanks!

### CODE REVIEW

**Comment by romartin** [comment]
ok to test

File: uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/ClientUserSystemManager.java [hunk 1]
```diff
@@ -276,6 +276,15 @@ public GroupManagerSettings getGroupManagerSettings() {
         return groupManagerSettings;
     }
 
+    /**
+     * Suggest if an exception have to be notified to th user.
```
Comment (dgutierr) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/ClientUserSystemManager.java]: Typo.

File: uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-api/src/main/java/org/uberfire/ext/security/management/api/exception/NoImplementationAvailableException.java [hunk 2]
```diff
@@ -27,4 +27,9 @@
 
     public NoImplementationAvailableException() {
     }
+
+    @Override
+    public String getMessage() {
+        return "No user/group security management provider available";
```
Comment (ederign) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-api/src/main/java/org/uberfire/ext/security/management/api/exception/NoImplementationAvailableException.java]: i18n?

Comment (ederign) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-api/src/main/java/org/uberfire/ext/security/management/api/exception/NoImplementationAvailableException.java]: (only if this goes to the user screen)

File: uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/ClientUserSystemManager.java [hunk 3]
```diff
@@ -85,8 +85,8 @@
         @Override
         public boolean error(final Message message,
                              final Throwable throwable) {
-            if (!(throwable instanceof NoImplementationAvailableException)) {
-                // If no user management services, do not complain. Just complain for service errors, if specified.
+            // If no user management services, do not complain. Just complain for service errors, if specified.
+            if (shouldDisplayError(throwable)) {
                 showError(throwable);
```
Comment (ederign) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/ClientUserSystemManager.java]: this show error method chances uses his own i18 message right?

File: uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-api/src/main/java/org/uberfire/ext/security/management/api/exception/NoImplementationAvailableException.java (hunk 2 above)
Comment (romartin) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-api/src/main/java/org/uberfire/ext/security/management/api/exception/NoImplementationAvailableException.java]: hey  @ederign , well yes, that exception messages are not presented to the user. But I added it to have a few more information on the logs

File: uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/ClientUserSystemManager.java (hunk 3 above)
Comment (romartin) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/ClientUserSystemManager.java]: yep

Comment (romartin) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/ClientUserSystemManager.java]: ah well mmmm, here is probably showing the exception's message itself, probably yeah, that's i18n missing here mmm.... lemme give it another look please :) Thx!

**Comment by romartin** [comment]
Hey @dgutierr @ederign 
Yeah looking deeper into the code found some duplication, some missing i18n's and other custom logic in different places... so finally added missing i18n messages and created the type `ClientSecurityExceptionMessageResolver` which manages all this stuff in a single place.
Thanks for the feedback! Let's keep improving this :)

**Comment by romartin** [comment]
Hey @dgutierr @ederign , just as a reminder, please do not forget to take a look at this one. Finally added all missing i18n's too, see comment above :)
Thanks!

**Pull request merged**